File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.py
def solution(n):
    
    count = [True] * (n+1)
    count[0] = count[1] = False
    
    for i in range(2, int(n**0.5)+1): #2부터 제곱근까지 소수를 구하기
        if count[i]:
            for j in range(i*i, n+1, i):
                count[j] = False
    
    return sum(count)

File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.js
function solution(n) {
    let count = new Array(n+1).fill(true);
    count[0] = count[1] = false;

    for (let i = 2; i <= Math.sqrt(n); i++) {
        if (count[i]) {
            for (let j = i*i; j <= n; j += i) {
                count[j] = false;
            }
        }
    }

    return count.filter(x => x).length;
}